### PR TITLE
Append WX_FLAVOUR to the name of wxrc executable

### PR DESCRIPTION
**Pull request by ardovm**
The name of the wxrc executable does not contain the "flavour" that was indicated to the configure script.

As the libraries are named according to the flavour, it is IMHO a good idea to let the tools follow the same rule.

FYI the [FreeBSD wxWidgets port](https://svnweb.freebsd.org/ports/head/x11-toolkits/wxgtk31/files/patch-utils-wxrc-Makefile.in?view=markup) has been marking the wxrc executable since a long time.

### CODE REVIEW

**Comment by ardovm** [comment]
The FreeBSD port, in fact:

- does not use just the "flavour", but rather `$(TOOLCHAIN_NAME)`;
- it [also changes](https://svnweb.freebsd.org/ports/head/x11-toolkits/wxgtk31/files/patch-Makefile.in?revision=472232&view=markup) the name of `wx-config` according to the same rule.

If you think either would be a good idea, I can modify this PR.

**Comment by vadz** [comment]
To be honest, I never quite understood what was the flavour supposed to be used for (the fact that it's not documented anywhere doesn't help, of course...), so I'm not sure at all if it's supposed to apply to executables too. I guess it doesn't matter much anyhow, as long as there is a flavour-less symlink anyhow, but I'll wait a bit to see if nobody else (perhaps there are even some people actually using `--with-flavour` option here?) has any objections.

**Comment by vadz** [comment]
So it looks like it all stems from [this bug](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=228904) i.e. is about installing wxGTK2 and wxGTK3 in parallel. I agree that it would be nice if this could be made to work, but is this enough for it?

Also, I wish OS package maintainers thought to contact us to discuss things like this, instead of just patching their versions...  If you are in contact with them, please ask them to consider doing this in the future.

**Comment by vadz** [comment]
I am still not sure what should we do about this, i.e. whether this is enough to allow installing wxGTK 2 and 3 in parallel or if more changes are needed. If anybody has the time to test this, it would be great (i.e. just configure and build using `--with-flavour=gtk2` and `--with-flavour=gtk3` in 2 different directories and run `make install` and check if both can be used).

**Comment by ardovm** [comment]
FWIW, the flavor is very useful for having different builds of wxWidgets coexist on the same machine. While we were working on a very long PR, we were using our "own build" of wxWidgets for the development of our own software, whereas "stock" versions of wxWidgets were installed from Ubuntu's repositories for other tools (wxPython, Audacity etc)
The flavor helped our Makefile's link to the correct (i.e. our) build just by calling the flavored wx-config script.

We also adopt flavors to have one debug-enabled build and a release build coexist together.

With respect to other possible uses of the flavors, I can also think of:
- having different versions (like 3.0, 3.1, 3.2...)
- having versions with different compile-time options (like gtk2/gtk3, wchar/utf8, debug/release...)
- when wxWidgets 2.8 were still around, having ANSI and Unicode builds.

If there are other ways to have multiple versions coexist seamlessly, then flavors could just be deprecated, IMHO.

**Comment by vadz** [comment]
Yes, I understand that it is useful, I'm just not sure if it actually works...

FWIW I never use it just because I never install all the multiple builds of wx I have and use them in-place, by invoking `wx-config` using the full path. But this doesn't work if you really want to have both of them globally available, of course, so flavour should still be useful.

**Comment by ardovm** [comment]
I've looked at the internals of the `wx-config` script and I found some quite interesting stuff in there.
It seems to already support multiple coexisting builds of wxWidgets. Quoting file `wx-config.in` at line 65:
```
    If multiple builds of wxWidgets  are available,  you can use the options
  --prefix, --host, --toolkit, --unicode, --static, --universal or --version
  to select from them.
```
(also `--flavour` is accepted as parameter, but it is not mentioned. I am going to commit a fix).

If I understand correctly, this is telling us that wxWidgets builds with Gtk2 and Gtk3 should _already_ be able to coexist, and they are chosen by invoking the common `wx-config` script with the corresponding `--toolkit` parameter.

In other words, flavors are not required (therefore not supposed to) allow Gtk2 vs Gtk3 builds, as well as all the other possible selectable options cited above.

On the other hand, the modification I proposed makes wx-config capable to select the "flavored" `wxrc` binary.

**Comment by vadz** [comment]
Yes, you can select different builds by their toolkit using `wx-config`, but selecting them by flavour is also supposed to work -- I'm just (still) not sure if it does.

Anyhow, I'll probably just merge this, it should do no harm, even if it's not perfect, thanks.

**Comment by ardovm** [comment]
Thank you very much!

**Comment by ardovm** [comment]
Sorry for resurrecting this PR.. I just wanted to remind you to kindly re-run bitbake as the Makefile.in in master remains the old one.
Thank you in advance!

**Comment by ardovm** [comment]
Never mind! I have to open another PR on wxrc. I'll do it there myself.
